### PR TITLE
CMR 9430: Fix SQL Injection Vulnerabilities in CMR Metadata App in provider and collection files

### PR DESCRIPTION
**Pull request by jmaeng72**
Fixed SQL injections in the provider and collection table files. There is a separate ticket (9431) that will fix the granule-table and concept files.

### CODE REVIEW

File: metadata-db-app/src/cmr/metadata_db/data/util.clj [hunk 1]
```diff
@@ -74,3 +74,9 @@
       (-> concept
           (assoc :metadata umm-json)
           (assoc :format (mt/format->mime-type :umm-json))))))
+
+(defn validate-table-name
+  "Checks if table name is valid, throws error if not valid"
+  [table-name]
+  (if (nil? (re-matches #"^[a-zA-Z0-9_]*$" table-name))
```
Comment (jaybarra) [inline metadata-db-app/src/cmr/metadata_db/data/util.clj:81]:
```suggestion
  (when-not (re-matches #"^[a-zA-Z0-9_]*$" table-name)
```

File: metadata-db-app/src/cmr/metadata_db/data/util.clj [hunk 2]
```diff
@@ -74,3 +74,9 @@
...
+    (throw (Exception. (str "Table name '" table-name "' is not valid. Can only have alphanumeric and underscore chars.")))))
```
Comment (jaybarra) [inline metadata-db-app/src/cmr/metadata_db/data/util.clj:82]:
```suggestion
  (when-not (re-matches #"^[a-zA-Z0-9_]*$" table-name)
    (throw (ex-info "Table name is not valid. Can only have alphanumeric and underscore chars." 
                    {:invalid-table table-name})))
```

File: metadata-db-app/test/cmr/metadata_db/test/test_utils.clj [hunk 3]
```diff
@@ -0,0 +1,6 @@
+(ns cmr.metadata-db.test.test-utils)
+
+(defn remove-spaces-and-new-lines
+  [input-str]
+  (clojure.string/replace (clojure.string/replace input-str #"\n" "") #" " "")
+  )
```
[the remark this answers is not in the document]
Comment (jceaser) [reply metadata-db-app/test/cmr/metadata_db/test/test_utils.clj:6]: those sound like good ideas, I would think that trim would be faster then regex.

File: metadata-db-app/src/cmr/metadata_db/data/oracle/collection_table.clj [hunk 4]
```diff
@@ -1,7 +1,8 @@
 (ns cmr.metadata-db.data.oracle.collection-table
   "Contains helper functions to create collection table."
   (:require
-   [clojure.java.jdbc :as j]))
+   [clojure.java.jdbc :as j]
```
Comment (jceaser) [inline metadata-db-app/src/cmr/metadata_db/data/oracle/collection_table.clj:4]: single letter imports is now discouraged by the project.

File: metadata-db-app/test/cmr/metadata_db/test/data/oracle/util.clj [hunk 5]
```diff
@@ -0,0 +1,12 @@
+(ns cmr.metadata-db.test.data.oracle.util
+  (:require
+    [clojure.test :refer :all]
+    [cmr.metadata-db.data.util :as util]))
+
+(deftest validate-table-name-test
+  (testing "correct table name given"
+    (are [table-name] (= (util/validate-table-name table-name))
```
Comment (jceaser) [inline metadata-db-app/test/cmr/metadata_db/test/data/oracle/util.clj:8]: switch to are3

File: metadata-db-app/src/cmr/metadata_db/data/oracle/collection_table.clj [hunk 6]
```diff
@@ -91,6 +94,7 @@
 
 (defmethod create-collection-indexes false
   [db _ table-name]
+  (util/validate-table-name table-name)
   (j/db-do-commands db (format "CREATE INDEX %s_crdi ON %s (concept_id, revision_id, deleted, delete_time)"
```
Comment (jceaser) [inline metadata-db-app/src/cmr/metadata_db/data/oracle/collection_table.clj:98]: we can't use j/db-do-prepared ?

File: metadata-db-app/src/cmr/metadata_db/data/util.clj (hunk 2 above)
Comment (jmaeng72) [reply metadata-db-app/src/cmr/metadata_db/data/util.clj:82]: Will make this change

File: metadata-db-app/test/cmr/metadata_db/test/test_utils.clj (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmaeng72) [reply metadata-db-app/test/cmr/metadata_db/test/test_utils.clj:6]: I think I'm just going to remove this help func. I originally put it in there to make the testing easier, but it's not necessary rn

**Review by jceaser — APPROVED** [review]

File: metadata-db-app/test/cmr/metadata_db/test/data/oracle/collection_table.clj [hunk 7]
```diff
@@ -0,0 +1,43 @@
+(ns cmr.metadata-db.test.data.oracle.collection-table
+  (:require
+    [clojure.test :refer :all]
+    [cmr.metadata-db.data.oracle.collection-table :as ct]
+    [cmr.metadata-db.test.test-utils :as tu]))
+
+(deftest collection-constraint-sql-false-test
+  (testing "valid table name"
+    (let [non-small-provider {:provider-id "PROV1", :short-name "test provider", :cmr-only false, :small false}]
+        (are [table-name query] (= query (tu/remove-spaces-and-new-lines (ct/collection-constraint-sql non-small-provider table-name)))
```
[the remark this answers is not in the document]
Comment (jmaeng72) [reply metadata-db-app/test/cmr/metadata_db/test/data/oracle/collection_table.clj]: done

File: metadata-db-app/src/cmr/metadata_db/data/oracle/collection_table.clj (hunk 4 above)
Comment (jmaeng72) [reply metadata-db-app/src/cmr/metadata_db/data/oracle/collection_table.clj]: done

**Review by zimzoom — APPROVED** [review]

**Pull request merged**